rust/operator/src/lib.rs

```diff
@@ -0,0 +1,674 @@
+        _id_mapping: &PodToNodeMapping,
+    ) -> Result<HashMap<&'static str, ConfigMap>, Error> {
+        Ok(HashMap::new())
+    }
+
+    /// Creates the pod required for the superset instance.
+    ///
+    /// # Arguments
+    ///
+    /// - `pod_id` - The `PodIdentity` containing app, instance, role, group names and the id.
+    /// - `node_name` - The node_name for this pod.
+    /// - `config_maps` - The config maps and respective types required for this pod.
+    /// - `validated_config` - The validated product config.
+    ///
+    async fn create_pod(
+        &self,
+        pod_id: &PodIdentity,
+        node_name: &str,
+        _config_maps: &HashMap<&'static str, ConfigMap>,
+        validated_config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
+    ) -> Result<Pod, Error> {
+        let version = &self.context.resource.spec.version;
+
+        let pod_name = name_utils::build_resource_name(
+            pod_id.app(),
+            &self.context.name(),
+            pod_id.role(),
+            Some(pod_id.group()),
+            Some(node_name),
+            None,
+        )?;
+
+        let annotations = BTreeMap::new();
+
+        let mut recommended_labels = get_recommended_labels(
+            &self.context.resource,
+            pod_id.app(),
+            &version.to_string(),
+            pod_id.role(),
+            pod_id.group(),
+        );
+        recommended_labels.insert(ID_LABEL.to_string(), pod_id.id().to_string());
+
+        let secret = validated_config
+            .get(&PropertyNameKind::Env)
+            .unwrap()
+            .get(CREDENTIALS_SECRET_PROPERTY)
+            .unwrap();
+
+        let container = Container {
+            name: String::from(APP_NAME),
+            image: Some(format!("{}:{}", IMAGE, version.to_string())),
+            env: Some(vec![
+                env_var_from_secret("SECRET_KEY", secret, "connections.secretKey"),
+                env_var_from_secret(
+                    "SQLALCHEMY_DATABASE_URI",
+                    secret,
+                    "connections.sqlalchemyDatabaseUri",
+                ),
+            ]),
+            ports: Some(vec![ContainerPort {
+                container_port: PORT,
+                name: Some(String::from(HTTP_PORT)),
+                ..Default::default()
+            }]),
+            ..Default::default()
+        };
+
+        let pod = PodBuilder::new()
+            .metadata(
+                ObjectMetaBuilder::new()
+                    .generate_name(pod_name)
+                    .namespace(&self.context.client.default_namespace)
+                    .with_labels(recommended_labels)
+                    .with_annotations(annotations)
+                    .ownerreference_from_resource(&self.context.resource, Some(true), Some(true))?
+                    .build()?,
+            )
+            .add_container(container)
+            .node_name(node_name)
+            .build()?;
```

Comment:
The name resolution is different in the host network. The instructions in the documentation would not work anymore because the name of the PostgreSQL service could not be resolved. I discussed again wit @lfrancke and we decided that the Superset pods are not created in the host network for now until we have really a use case which makes this necessary.